core/services/workflows/engine.go

```diff
@@ -775,11 +775,8 @@ func (e *Engine) configForStep(ctx context.Context, executionID string, step *st
 		return step.config, nil
 	}
 
-	// Merge the configs for now; note that this means that a workflow can override
-	// all of the config set by the capability. This is probably not desirable in
-	// the long-term, but we don't know much about those use cases so stick to a simpler
-	// implementation for now.
-	return merge(capConfig.DefaultConfig, step.config), nil
+	// Merge the configs for now with registry config overriding the step config.
```

Comment:
Can we add an explaining for why?

Eg: "some config fields are sensitive and could affect the safe running of the capability, so we want to avoid users being able to modify this. To accomplish this, the capabilities registry overrides what the user provides."